include/taskolib/Step.h

```diff
@@ -139,10 +148,12 @@ class Step
      *            inappropriate return value for the step type (see above), if a timeout
      *            is encountered, or if termination has been requested explicitly by the
      *            script.
+     *
+     * \see For more information about step setup scripts see at Sequence.
      */
-    bool execute(Context& context)
+    bool execute(Context& context, std::string step_setup = "")
```

Comment:
I wonder if the `step_setup` script is not - from a logic perspective - a part of the `Context`?

Here we add a new parameter to multiple functions.

Reply:
IMO we have two conceptual levels:
- Everything that the end user can edit, modify etc. is in the `Sequence`
- Everything that is only of interest to the server developer is in the `Context`

So I think it makes sense to put the setup script into `Sequence`.

Reply:
That is correct, partially, I believe.

The `step_setup_script` is part of the `Sequence` and not of the `Context` that the user hands over :heavy_check_mark: 

But how to transfer the extra information from the `Sequence` to the `Step`.
I believe looking from the `Step`'s point of view the `step_setup_script` is part of the `Context` it gets.

If we introduce a new parameter (like done here), that will complicate the intra-Step API for no real reason. Imagine we add another conceptually equal thing, maybe Sequence Color. Shall we then add yet another parameter, or shall all the meta information be stuffed into the Context?
Of course the USER shall not put the `step_setup_script` into the `Context.step_setup_script`, but the `Sequence` can store it there if it executes a `Step`. For example the `Context.step_setup_script` is protected and `Sequence` and `Step` are friends.

Reply:
> But how to transfer the extra information from the `Sequence` to the `Step`. I believe looking from the `Step`'s point of view the `step_setup_script` is part of the `Context` it gets.
> 
> If we introduce a new parameter (like done here), that will complicate the intra-Step API for no real reason. Imagine we add another conceptually equal thing, maybe Sequence Color. Shall we then add yet another parameter, or shall all the meta information be stuffed into the Context? Of course the USER shall not put the `step_setup_script` into the `Context.step_setup_script`, but the `Sequence` can store it there if it executes a `Step`. For example the `Context.step_setup_script` is protected and `Sequence` and `Step` are friends.

Good point. I also don't like the proliferation of more and more function parameters. I am not a friend of `friend` either, though (pun intended). But as I do not have a better idea, maybe we should try that?

Reply:
We could `public` the data member and make it an ordinary part of the `Context`, so no need to be friends.
Further we document that `Sequence` will overwrite that in the `Context` on execution (if the `Step` is executed via a `Sequence`, of course, only).

In this way it will become very simple to test, (protected stuff is always ugly).

That's a documentation 'issue' then. Do we believe people are able to see that they should not set it, because it will be in vain anyhow (on `Sequence` execution)?

Reply:
> We could `public` the data member and make it an ordinary part of the `Context`, so no need to be friends.
> Further we document that `Sequence` will overwrite that in the `Context` on execution (if the `Step` is executed via a `Sequence`, of course, only).

I can live with that. :+1:

Reply:
Moved `step_setup` to `Context`.